src/Neves/Events/TransactionalDispatcher.php

```diff
@@ -146,6 +146,12 @@ protected function addPendingEvent($connection, $event, $payload)
         $connectionId = $connection->getName();
         $transactionLevel = $this->transactionLevel;
 
+        // When working with multiple connections, we might come across
+        // one that isn't prepared yet, nor will it have any events
+        if (! array_key_exists($connectionId, $this->pendingEvents)) {
```

Comment:
We can replace this with `isset($this->pendingEvents[$connectionId])`.

Reply:
Also, please move this to a private method that checks if a connection: `isPrepared($connection)`.

So, the logic can be reused in other methods, and it will be.